### PR TITLE
Add comment and semicolon tokens to source_file$parsed_content

### PR DESCRIPTION
**Pull request by fangly**
Hi Jim,
This is my attempt to fix the issue described at #215. The key change was a small modification of the `top_level_expressions()` function so that it returns also the ID of "COMMENT" and "';'" tokens. The rest is just adapting the linters to this change. All tests succeed.
What do you think?
Best,
Florent

### CODE REVIEW

File: R/get_source_expressions.R [hunk 1]
```diff
@@ -309,12 +308,13 @@ next_with_parent <- function(pc, loc) {
...
   loc <- which(with_parent$id == id)
 
-  return(which(pc$id == with_parent$id[loc + 1L]))
+  which(pc$id == with_parent$id[loc + 1L])
 }
 
 top_level_expressions <- function(pc) {
   if (is.null(pc)) {
-    return(NULL)
+    character()
```
Comment (jimhester) [inline R/get_source_expressions.R]: This function is returning the output of `which()` from the other branch, which is a integer vector, so this should be `integer()`. You are correct the previous return value of `NULL` was incorrect.

File: R/get_source_expressions.R [hunk 2]
```diff
@@ -309,12 +308,13 @@ next_with_parent <- function(pc, loc) {
...
+  } else {
```
Comment (jimhester) [inline R/get_source_expressions.R]: I would prefer not to have an else here

File: R/utils.R [hunk 3]
```diff
@@ -76,14 +77,12 @@ quoted_blanks <- function(matches, shift_start = 0, shift_end = 0) {
...
-  loc <- which(fun(source_file$parsed_content$token, value))
-  if (loc %==% integer(0)) {
-    NULL
-  } else {
+  pc <- source_file$parsed_content
+  if (!pc$col1 %==% integer(0) &&
+      !(loc <- which(fun(pc$token, value))) %==% integer(0) ) {
     loc
+  } else {
+    character()
```
Comment (jimhester) [inline R/utils.R]: Again this should be `integer()`

File: R/utils.R [hunk 4]
```diff
@@ -76,14 +77,12 @@ quoted_blanks <- function(matches, shift_start = 0, shift_end = 0) {
...
-  if (source_file$parsed_content$col1 %==% integer(0)) {
-    return(NULL)
-  }
-  loc <- which(fun(source_file$parsed_content$token, value))
-  if (loc %==% integer(0)) {
-    NULL
-  } else {
+  pc <- source_file$parsed_content
+  if (!pc$col1 %==% integer(0) &&
+      !(loc <- which(fun(pc$token, value))) %==% integer(0) ) {
```
Comment (jimhester) [inline R/utils.R]: Please don't use assignments within conditionals

**Comment by fangly** [comment]
Jim, good point about the ids being integers. I always think of them as strings. I've made the appropriate changes.

**Comment by jimhester** [comment]
Thanks!

**Pull request merged**